test/util/test_function_util.py

```
@@ -250,8 +255,176 @@ def test_update_function_declaration_at_bottom(copy_file, remove_file) -> None:
+        spec_partition_7,
+        spec_partition_9,
+        spec_partition_10,
+    ]
+    expected_spec = spec_partition_canonical
+    for spec in specs:
+        normalized_spec = normalize_function_specification(spec)
+        if normalized_spec != expected_spec:
+            msg = f"Actual:\n{normalized_spec}\nExpected:\n{expected_spec}"
+            if normalized_spec.eq_setwise(expected_spec):
+                msg = f"Same clauses, but different order.\n{msg}"
+                warnings.warn(msg)
```

Ruff (0.14.13)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)